test-interface/src/main/scala/scala/scalanative/testinterface/TestMain.scala

```diff
@@ -31,53 +31,27 @@ object TestMain {
   final val iPv4Loopback = "127.0.0.1"
   final val iPv6Loopback = "::1"
 
-  private def getFreeBSDLoopbackAddr(): String = {
+  private def setFreeBSDWorkaround(): Unit = {
     /* Standard out-of-the-box FreeBSD differs from Linux & macOS in
      * not allowing IPv4 mapped IPv6 addresses, such as :FFFF:127.0.0.1
-     * or ::ffff:7f00:1.  These can be enabled by setting the line
-     * ipv6_ipv4mapping="YES" in /etc/rc.conf (and rebooting).
+     * or ::ffff:7f00:1.  These can be enabled at runtime by running
      *
-     * FreeBSD TestMain initial connections should succeed on both IPv4
-     * and IPv6 systems without requiring arcane and non-standard system
-     * configuration.  This method checks the protocol that Java connect()
-     * is likely used and returns the corresponding loopback address.
-     * Tests which use IPv4 addresses, either through hard-coding or
-     * bind resolution, on IPv6 systems will still fail. This allows to
-     * run the vast majority of tests which do not have this characteristic.
+     *   sysctl net.inet6.ip6.v6only=0
```

Comment:
s/by running/by previously running/  

or such: i.e. no concurrent modification

Reply:
Hmmm, what do you mean with concurrent modification?

Reply:
I am suggesting making the timing of the "sysctl" explicit.  That is, before TestMain is run.
That should give the desired results. i.e. sysctl & TestMain do not run at the same time, probably
in different processes.

As you may know better than I, sometimes changing a sysctl value will take effect immediately while a 
program is executing and sometimes only the next time it is run.  I suspect that changing ip6.v6only
"on-the-fly" will either have no effect or mess TestMain up totally.  

Yes, "on-the-fly is allowed" is a perverse interpretation of the existing text. Since the existing text
does not rule it out, somebody is bound to try it. Then again, even with it being ruled out, somebody
is bound to try it ;-)